File: Bot.java

public class Bot {
    int round;
    int coins;
    int bank;
    String memory;

    public Bot() {
        round = 0;
        coins = 0;
        bank = 0;
        memory = "cooperate";}

    public String toString() {
        return getClass().getName() + String.format("(%d)",this.bank);}

    public void reset(){
        this.round = 0;
        this.coins = 0;
        this.memory = "cooperate";
        secondary_reset();}

    public void secondary_reset() {
    }

    public String decide() {
        return "cooperate";
    }
}

File: bot.py
class Bot (object):
    def __init__(self):
        self.reset()

    def reset(self):
        self.round = 0
        self.coins = 0
        self.bank = 0
        self.memory = 'cooperate'
        self.secondary_reset()
        return self

    def secondary_reset(self):
        pass

    @classmethod
    def spawn(cls):
        "Return new copy of self"
        return cls()

    def __repr__(self):
        return f"{type(self).__name__}({self.bank})"
